File: cli/testdata/sdk_resolve/create_widget.ts
import { intent, resource, stack } from "@ubx/sdk";
import { FakeWidget } from "./bindings.ts";

export default stack("payments", () => {
  intent({ summary: "add widget1, authored in TypeScript" });
  resource(FakeWidget, "widget1", { name: "widget1", tags: { env: "prod" } });
});

File: cli/testdata/sdk_resolve_go/create_widget.go
// Mirrors cli/testdata/sdk_resolve/create_widget.ts exactly, in Go.
package main

import sdk "github.com/ubiquex/ubx-sdk-go/runtime"

func main() {
	sdk.Main(sdk.Stack("payments", func() {
		sdk.Intent(sdk.IntentInfo{Summary: "add widget1, authored in Go"})
		sdk.Resource(FakeWidget, "widget1", FakeWidgetConfig{
			Name: "widget1",
			Tags: map[string]any{"env": "prod"},
		})
	}))
}
